src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -24,6 +24,7 @@
 namespace clipper {
 
 const std::string LOGGING_TAG_TASK_EXECUTOR = "TASKEXECUTOR";
+constexpr size_t CACHE_SIZE_ELEMENTS = 2048;
```

Comment:
Because cache entries are of variable size and the user is unaware of the size of entry metadata, I think this should be a configurable value named `CACHE_SIZE_BYTES` instead. However, making this change complicates cache eviction substantially and will reduce CLOCK performance. Let's discuss @dcrankshaw.